TritonSE/TAP-Foster-Dog-Matching
# Application Connect Backend

### Administrative Info
[GitHub Issue #]
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
What changes did you make?

Foster side:
- Form now creates application in backend for currently logged in user
- When you click back from the agreement form, the data you filled out in the previous form persists
- Prevented repetitive application creation (can't spam the submit button to create new applications. will just update existing application after one click)

Admin side:
- Made the form populate itself based on an input application-id's fetched data
- Disabled any edits to the values of the form + no submit app button 

Note: I had to make a the ambassador, coordinator, selectedDogs, and preference fields optional to make this work. I also set the completedActionItems to false and status to "Step 1: Application" in order to account for this incomplete application state.


### Testing
You can test by doing the following:

Foster side:
 - Check that you can go back to previous page and progress is saved for the initial long info form 
 - Fill out the form and submit, check that it shows up in the backend by searching through the mongo entries (or uncomment line 577 to check status of create app request + uncomment line 571, 572 to check status of update app request)
 - Make sure user can't create the same application repetitively 

Admin side:
- Check that none of the fields are modifiable
- [Optional]: Manually change the application id on line 641 to whatever application you want to display (Note: some of the older applications in the database conform to slightly different models, so if you pick an older application, a few fields might not be visible)
